index.js

```diff
@@ -147,6 +147,18 @@ function fastifyMultipart (fastify, options, done) {
           }
         }
       }
+      if (options.attachFieldsToBody === 'keyValues') {
+        const body = {}
+        for (const key of Object.keys(req.body)) {
+          const field = req.body[key]
+          if (field.value !== undefined) {
+            body[key] = field.value
+          } else if (field._buf !== undefined) {
+            body[key] = field._buf.toString()
```

Comment:
I don't think it is a good idea for `.toString()` a buffer directly.

Reply:
Since, it is `buffer` and people usually upload photo using `form`.
I would probably encode it as `base64`.

Reply:
I dont know if this is the best way to do it... either keep it as a buffer or encode it to a binary string. Encoding to a base64 string would just result 30% memory overhead and you would still need to decode it base64 to make something useful with it.

Reply:
Keep it as `buffer` will invalidate by the `validation` phase which do not match the goal of this PR.
Put the whole file buffer inside memory is not a good deal already. I would accept even more consumption in this case.

For `binary` string, it can be an option. But it needs to document how to transform it back to `buffer` or how to use it with `stream` or `fs.writeFile`, etc.

Reply:
> people usually upload photo using form

It seems like for a photo upload use-case, the `attachFieldsToBody` option wouldn't be used, as it buffers the entire photo in memory which could be quite large. If people still want to do that, then can use `attachFieldsToBody: true`. This PR does not _remove_ that existing option so it will no have no impact on that use case.

A use-case that I'd like to support is curl commands which can include key values directly in the command, as well as point to files for the value, e.g.

```shell
curl -i -X POST http://localhost:1234/auth \
  -F "username=matt" \
  -F "key=@privatekey.txt" \
  -F "thing=123"
```

Additionally, the `buffer.toString()` behavior can be bypassed / customized using the existing `onFile` handler. I've included an example of using it to decode base64 in the unit tests:

https://github.com/fastify/fastify-multipart/blob/8eaf5f303ea59be087b5dbeb62b40b42894c9c8e/test/multipart-attach-body.test.js#L115-L120